extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1442,6 +1477,81 @@ export class Project implements ISqlProject {
+	 * @param caseSensitive Determines if value to add to collection should be compared against existing values case sensitively
+	 */
+	private async addValueToCollectionProjectProperty(propertyName: string, valueToAdd: string, caseSensitive?: boolean): Promise<void> {
+		if (valueToAdd.includes(';')) {
+			return;
+		}
+
+		let propertyValueToSet = this.evaluateProjectPropertyValue(propertyName);
+		if (propertyValueToSet === undefined) {
+			propertyValueToSet = valueToAdd;
+		} else {
+			const propertyValue = caseSensitive ? propertyValueToSet : propertyValueToSet.toLocaleLowerCase();
+			const originalValueToAdd = valueToAdd;
+			valueToAdd = caseSensitive ? valueToAdd : valueToAdd.toLocaleLowerCase();
+			if ((propertyValue.length > valueToAdd.length
+				&& (propertyValue.startsWith(`${valueToAdd};`)
+					|| propertyValue.endsWith(`;${valueToAdd}`)
+					|| propertyValue.indexOf(`;${valueToAdd};`) >= 0))
+				|| (propertyValue.length === valueToAdd.length
+					&& propertyValue === valueToAdd)) {
+				return;
+			}
+
+			propertyValueToSet = `${propertyValueToSet};${originalValueToAdd}`;
+		}
+
+		await this.setProjectPropertyValue(propertyName, propertyValueToSet);
+	}
+
+	/**
+	 * Removes value from project property where values are semicolon separated
+	 * @param propertyName Name of property
+	 * @param valueToRemove Value to remove from the project property
+	 * @param caseSensitive Determines if value to remove from collection should be compared case sensitively
+	 */
+	protected async removeValueFromCollectionProjectProperty(propertyName: string, valueToRemove: string, caseSensitive?: boolean): Promise<void> {
+		let propertyValueToSet = this.evaluateProjectPropertyValue(propertyName);
+		if (propertyValueToSet === undefined
+			|| propertyValueToSet.length < valueToRemove.length
+			|| valueToRemove.includes(';')) {
+			return;
+		}
+
+		const propertyValue = caseSensitive ? propertyValueToSet : propertyValueToSet.toLowerCase();
+		valueToRemove = caseSensitive ? valueToRemove : valueToRemove.toLowerCase();
+
+		if (propertyValue.length > valueToRemove.length) {
+			let valueToRemovePosition: number;
+
+			if (propertyValue.startsWith(`${valueToRemove};`)) {
+				propertyValueToSet = propertyValueToSet.substring(valueToRemove.length + 1);
+			} else if (propertyValue.endsWith(`;${valueToRemove}`)) {
+				propertyValueToSet = propertyValueToSet.substring(0, propertyValue.length - valueToRemove.length - 1);
+			} else if ((valueToRemovePosition = propertyValue.indexOf(`;${valueToRemove};`)) >= 0) {
+				propertyValueToSet = propertyValueToSet.substring(0, valueToRemovePosition + 1)
+					+ propertyValueToSet.substring(valueToRemovePosition + valueToRemove.length + 2);
+			} else {
+				return;
+			}
+
+			await this.setProjectPropertyValue(
+				propertyName,
+				propertyValueToSet);
+		} else if (propertyValue.length === valueToRemove.length
+			&& propertyValue === valueToRemove) {
```

Comment:
just this condition is sufficient; let the JS runtime decide whether the length comparison optimization is worthwhile